apps/nextjs-example/pages/index.tsx

```diff
@@ -340,6 +368,109 @@ function OptionalFunctionality() {
 
+    const onSubmitScript = async () => {
+        const payload = {
+            type: "script_payload",
+            code: {
+                bytecode: DO_NOTHING_SCRIPT
+            },
+            type_arguments: [],
+            arguments: []
+        };
+
+        const response = await signAndSubmitTransaction(payload);
+        try {
+            if (response?.hash === undefined) {
+                throw new Error(`No response given ${response}`)
+            }
+            await aptosClient(network?.name.toLowerCase()).waitForTransaction(response.hash);
+            setSuccessAlertHash(response.hash, network?.name);
+        } catch (error) {
+            console.error(error);
+        }
+        setSuccessAlertMessage(
+            JSON.stringify({signAndSubmitTransaction: response})
+        );
+    };
+
+    // TODO: Let's put this into a cookie or local storage so it only happens once
+    const createFeePayerAccount = async (client: AptosClient) => {
+        // Already exists, so lets not fund it
+        if (feePayerAccount) {
+            return feePayerAccount;
+        }
+
+        let newAccount = new AptosAccount();
+        let feePayerAddress = newAccount.address();
+
+        try {
+            let coinClient = new CoinClient(client);
+            let balance = await coinClient.checkBalance(feePayerAddress);
+            if (Number(balance) >= 100000000) {
+                return newAccount;
+            }
+        } catch (error: any) {
+            // Swallow errors, likely due to account not existing
+        }
+        await (faucet(network?.name.toLowerCase()).fundAccount(feePayerAddress, 100000000));
+
+        return newAccount;
+    }
+
+    const onSubmitFeePayer = async () => {
+        if (!account) {
+            throw new Error("Not connected");
+        }
+
+        // TODO: MultiSig isn't supported here properly
+        if (typeof account.minKeysRequired !== undefined) {
+            throw new Error("MultiSig not supported");
+        }
+
+        let provider = aptosClient(network?.name.toLowerCase());
+        // Generate an account and fund it
+        let feePayerAccount = await createFeePayerAccount(provider);
+
+        const payload: Types.TransactionPayload = {
+            type: "entry_function_payload",
+            function: "0x1::aptos_account::transfer",
+            type_arguments: [],
+            arguments: [account.address, 1], // 1 is in Octas
+        };
+
+        // This would normally be provided by the service or a server
+        // We need to increase the default timeout
+        let expiration_timestamp_secs = Math.floor(Date.now() / 1000) + 60000;
+
+        const rawTxn = await provider.generateFeePayerTransaction(account.address, payload, feePayerAccount.address().hex(), [], {expiration_timestamp_secs: expiration_timestamp_secs.toString()})
+
+        // Sign with fee payer
+        const feePayerAuthenticator = await provider.signMultiTransaction(feePayerAccount, rawTxn);
```

Comment:
I'm confused about `signMultiTransaction`, looking at the implementation [here](https://github.com/aptos-labs/aptos-core/blob/a2877543f2a2a780919f341bd11169fd60ea522e/ecosystem/typescript/sdk/src/providers/aptos_client.ts#L382) there's nothing that should tie this function to `FeePayer` or `MultiAgent` as `TransactionBuilder.getSigningMessage` accepts all kinds of raw transactions

Reply:
This is maybe more of a feedback for the Aptos client

Reply:
Ok I guess that the returned `AccountAuthenticator` is only used as input for authenticators of a multi-agent or a fee-payer transaction.